### DISCUSSION TITLE
Does TritonRoute support all design rules in  ISPD 2014/2015 tech files?

### DISCUSSION THREAD

**Discussion by liangrj2014** [The-OpenROAD-Project/OpenROAD#3846] [Q&A] [answered]
We wonder whether TritonRoute support all design rules in ISPD 2014/2015 tech files (https://www.ispd.cc/contests/14/web/benchmarks.html)? Example design rules are as follows:

1. PROPERTY LEF57_MINSTEP "MINSTEP 0.100 MAXEDGES 1 ;" ;
2. MINIMUMCUT 2 WIDTH 0.400 ;
3. MINENCLOSEDAREA 0.30 ;
4. PROPERTY LEF57_SPACING "SPACING 0.13 PARALLELOVERLAP ;”;

**Comment by maliberty** [comment]
We test with ispd 18&19 but not those.  2 is only relevant for pdn and I know we support 3.  @osamahammad21 would you check on 1 & 4

**Comment by refaay** [comment]
We currently parse for LEF58 rules. We don't parse LEF57 (by looking at /src/odb/src/lefin/lefin.cpp and /src/drt/src/io/io.cpp).
I will start a PR for parsing LEF57.

Also, we don't support PARALLELOVERLAP option for LEF58_SPACING in drt according to io.cpp:

```
case odb::dbTechLayerCutSpacingRule::CutSpacingType::PARALLELOVERLAP:
        logger_->warn(utl::DRT,
                      260,
                      "Unsupported LEF58_SPACING rule for layer {} of type "
                      "PARALLELOVERLAP.",
                      layer->getName());
        break;
```

We support MAXEDGES for LEF58_MINSTEP as in io.cpp:

```
for (auto rule : layer->getTechLayerMinStepRules()) {
    auto con = make_unique<frLef58MinStepConstraint>();
    con->setMinStepLength(rule->getMinStepLength());
    con->setMaxEdges(rule->isMaxEdgesValid() ? rule->getMaxEdges() : -1);
    con->setMinAdjacentLength(
        rule->isMinAdjLength1Valid() ? rule->getMinAdjLength1() : -1);
    con->setEolWidth(rule->isNoBetweenEol() ? rule->getEolWidth() : -1);
    tmpLayer->addLef58MinStepConstraint(con.get());
    tech_->addUConstraint(std::move(con));
  }
```

**Reply by liangrj2014** [reply]
Got it! Thanks for the clarification!

**Comment by maliberty** [comment]
We could add support for new rules if there is strong interest

**Comment by refaay** [comment, marked as answer]
Hi @liangrj2014 , we added support for PROPERTY LEF57_MINSTEP & PROPERTY LEF57_SPACING.
Also, support for PARALLELOVERLAP will be added soon.

**Reply by liangrj2014** [reply]
Hi OpenROAD team! Really appreciate the great support!

**Comment by refaay** [comment]
@liangrj2014 PARALLELOVERLAP is now supported for LEF57_SPACING &LEF58_SPACING properties.
